frontend/src/common/HelpDialog.tsx

```diff
@@ -0,0 +1,37 @@
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline'
+import { Button, Dialog, DialogContent, DialogTitle, Slide, Typography } from '@mui/material'
+import { TransitionProps } from '@mui/material/transitions'
+import React, { ReactNode } from 'react'
+
+const Transition = React.forwardRef(function Transition(
+  props: TransitionProps & {
+    children: React.ReactElement<any, any>
+  },
+  ref: React.Ref<unknown>,
+) {
+  return <Slide direction='up' ref={ref} {...props} />
+})
+
+type Props = {
+  infoLabel: string
+  dialogTitle: string
+  content: ReactNode
+  icon?: ReactNode
+}
+
+function HelpDialog({ infoLabel, dialogTitle, content, icon }: Props) {
+  const [open, setOpen] = React.useState(false)
+
+  return (
+    <>
+      <Button size='small' onClick={() => setOpen(true)} endIcon={icon ? icon : <HelpOutlineIcon />}>
+        <Typography variant='caption'>{infoLabel}</Typography>
+      </Button>
+      <Dialog open={open} onClose={() => setOpen(false)} maxWidth='md' TransitionComponent={Transition}>
```

Comment:
It feels a bit out of place to be using a transition when we don't for any other dialog. I believe we should be consistent and either use a transition for all of them or none of them.

Reply:
I feel the same but also wanted to do all the others outside of this ticket. I'll remove it from this PR and make a new ticket to add transitions to all other dialogs because I do quite like it!